src/main/webapp/WEB-INF/view/profile.jsp

```diff
@@ -48,6 +71,24 @@ TextProcessor processor = BBProcessorFactory.getInstance().createFromResource("k
     </form>
      <% } %>
 
+     <hr/>
+     <h2><%= user.getName()%>'s Sent Messages</h2>
+     
+     <div id="activity">
+      <ul>
+     <%
+      for (Message message : messages) {
+        String author = UserStore.getInstance().getUser(message.getAuthorId()).getName();
```

Comment:
I think you can avoid this lookup. Isn't message.getAuthorId() always equivalent to user.getId()? If so, I think you could just use user.getName() here instead.

Also consider moving the messages declaration to right before the loop. I think that would make it easier to see that the variable contains only messages for user.getId() and would add extra confidence that user.getName() is appropriate here.